### ISSUE TITLE
Extend Readme

### ISSUE THREAD

**Issue by andheiberg** [RickWong/react-isomorphic-starterkit#119]
Hi guys,

Great to see this. I've dabbled in react when it came out, but haven't played with it since.

Wanted to find a repo that follows best practices and hopefully skip the pain of making all the rookie mistakes as I get back into it.

It would be great if you guys could write about why each component was chosen somewhere. A huge amount of information can be learned from knowing what each package is trying to solve and why you choose it.

Is this something you guys would be interested in adding?

**Comment by andheiberg** [comment]
Avatar.js uses withStyles but Main.js uses InlineCss. Is it not possible to only withStyles? I like having styles in separate files.

I see that Transmit does the default export so maybe that's why you can't use withStyles?

**Comment by andheiberg** [comment]
Reading https://github.com/RickWong/react-inline-css#sass--less there's a way to use InlineCSS with a seperate file, but again what's the value tradeoff between InlinceCss and withStyles?

**Comment by andheiberg** [comment]
Why is Koa-Proxy used?

**Comment by RickWong** [comment]
Koa-proxy is used to proxy GitHub API requests from the client in the example, so that the client-side never directly speaks to GitHub.

**Comment by RickWong** [comment]
The latest master commit is kind of a work in progress. I've played with a few isomorphic SASS setups and the most productive one would be a webpack sass+css-loaders and InlineCss, without style-loader or isomorphic-styleloader. I'll push that solution when it's more cleaned up.

The result won't use withStyles.

**Comment by RickWong** [comment]
And yes the readme can definitely explain more about the directory structure and some isomorphic concepts like external API access.

**Comment by andheiberg** [comment]
Yeah I can see what Koa-proxy achieves, but why do you not want the frontend to talk to github?

**Comment by andheiberg** [comment]
Cool I will keep an eye out, but yeah would be great to read more about the choices.

**Comment by hugotox** [comment]
Hi, is this project still alive?
